lib/ansible/modules/crypto/x509_crl.py

```diff
@@ -0,0 +1,791 @@
+            except ValueError:
+                pass
+
+        raise CRLError(
+            'The time spec "%s" for %s is invalid' %
+            (input_string, input_name)
+        )
+
+    def __init__(self, module):
+        super(CRL, self).__init__(
+            module.params['path'],
+            module.params['state'],
+            module.params['force'],
+            module.check_mode
+        )
+
+        self.update = module.params['mode'] == 'update'
+
+        self.privatekey_path = module.params['privatekey_path']
+        self.privatekey_passphrase = module.params['privatekey_passphrase']
+
+        self.issuer = crypto_utils.parse_name_field(module.params['issuer'])
+        self.issuer = [(entry[0], entry[1]) for entry in self.issuer if entry[1]]
+
+        self.last_update = self.get_relative_time_option(module.params['last_update'], 'last_update')
+        self.next_update = self.get_relative_time_option(module.params['next_update'], 'next_update')
+
+        self.digest = crypto_utils.select_message_digest(module.params['digest'])
+        if self.digest is None:
+            raise CRLError('The digest "{0}" is not supported'.format(module.params['digest']))
+
+        self.revoked_certificates = []
+        for i, rc in enumerate(module.params['revoked_certificates']):
+            result = {
+                'serial_number': None,
+                'revocation_date': None,
+                'issuer': None,
+                'issuer_critical': False,
+                'reason': None,
+                'reason_critical': False,
+                'invalidity_date': None,
+                'invalidity_date_critical': False,
+            }
+            path_prefix = 'revoked_certificates[{0}].'.format(i)
+            if rc['path'] is not None:
+                # Load certificate from file
+                if rc['serial_number'] is not None:
+                    module.fail_json(
+                        msg='{0}serial_number must not be specified if {0}path is specified'.format(path_prefix)
+                    )
+                if rc['issuer'] is not None:
+                    module.fail_json(
+                        msg='{0}issuer must not be specified if {0}path is specified'.format(path_prefix)
+                    )
+                try:
+                    cert = crypto_utils.load_certificate(rc['path'], backend='cryptography')
+                    try:
+                        result['serial_number'] = cert.serial_number
+                    except AttributeError:
+                        # The property was called "serial" before cryptography 1.4
+                        result['serial_number'] = cert.serial
+                    try:
+                        ext = cert.extensions.get_extension_for_class(x509.AuthorityKeyIdentifier)
+                        if ext.value.authority_cert_issuer is not None:
+                            result['issuer'] = list(ext.value.authority_cert_issuer)
```

Comment:
I realize now that this is probably incorrect. The `issuer` field should be set to the certificate's issuer, not to the issuer's issuer. I don't know how to convert the certificate's issuer (a [Name](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Name) object) to a list of [GeneralName](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.GeneralName) objects, though. So no idea how to improve on this. (This already puzzled me about [AuthorityKeyIdentifier](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.AuthorityKeyIdentifier)'s [authority_cert_issuer](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.AuthorityKeyIdentifier.authority_cert_issuer) -- it has the same "conversion problem".)

Reply:
Most X.509 attribute types use DirectoryName (specifically with UTF8String or PrinteableString - referring to the ASN.1 choices of directory name here, not python/crypto classes), in Python the DirectoryName is an instance of GeneralName.

Edge cases would probably be a nightmare though, technically older or noncompliant certificates could have values that weren't clean utf8/printable strings.

Reply:
Deleted my comment as being possible incorrect.

Reply:
Maybe it's best to just remove this functionality and let `issuer` only be entered manually by the user. We can re-add this later if we find out how to do it correctly :)

Reply:
I've changed the PR accordingly in bef34a7d71f7b25696a1100ffe35d88e3ce8051d